### ISSUE TITLE
In Editor Checkout: remove "Modal" from modal title

### ISSUE THREAD

**Issue by ciampo** [Automattic/wp-calypso#47829]
## What

The In Editor Checkout modal has a modal title that currently reads "Checkout modal". This title is not visible, but is accessible to screen readers.

The In Editor Checkout modal's title should be updated to "Checkout".

## Why

This change improves clarity, by removing the redundant "modal" from the modal's title.

## Acceptance Criteria

- [ ] The In Editor Checkout modal's title should read "Checkout" (it's currently "Checkout modal")

## Context

- as discussed in https://github.com/Automattic/wp-calypso/pull/47601#discussion_r531284368
- file: `client/blocks/editor-checkout-modal/index.tsx`

**Comment by worldomonation** [comment]
Hi! I'm performing triage on older bugs as part of Quality Squad's efforts to cut down on backlog issue counts.

I am hesitant to close accessibility related issues as stale, especially an issue like this which appear to be self-contained and small in size.

I would like to propose tagging this issue with a Good First Issue label and have this triaged by the Luna team. How does that sound?

**Comment by ciampo** [comment]
cc @razvanpapadopol
